src/error.rs

```diff
@@ -53,6 +54,8 @@ impl<'tcx> Error for EvalError<'tcx> {
         match *self {
             EvalError::FunctionPointerTyMismatch(..) =>
                 "tried to call a function through a function pointer of a different type",
+            EvalError::ZstPointerDeref =>
+                "tried to deref a pointer to the ZST allocation",
```

Comment:
the fact that accessing `&[u64; 0]` requires 8 byte alignment, or at least accessing it through `&[u64]` requires that. We never read from it, but we obtain the handle to that memory through `get` and `get_mut`